tests/SpecTests/CrudSpecTest.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests;
+
+use stdClass;
+
+/**
+ * Transactions spec tests.
+ *
+ * @see https://github.com/mongodb/specifications/tree/master/source/crid
+ */
+class CrudSpecTest extends FunctionalTestCase
+{
+    /* These should all pass before the driver can be considered compatible with
+     * MongoDB 4.2. */
+    private static $incompleteTests = [
+        'aggregate-merge: Aggregate with $merge' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and batch size of 0' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and majority readConcern' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and local readConcern' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and available readConcern' => 'PHPLIB-438',
+        'aggregate-out-readConcern: readConcern majority with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: readConcern local with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: readConcern available with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: readConcern linearizable with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: invalid readConcern with out stage' => 'PHPLIB-431',
+        'bulkWrite-arrayFilters: BulkWrite with arrayFilters' => 'Fails due to command assertions',
+        'updateWithPipelines: UpdateOne using pipelines' => 'PHPLIB-418',
+        'updateWithPipelines: UpdateMany using pipelines' => 'PHPLIB-418',
+        'updateWithPipelines: FindOneAndUpdate using pipelines' => 'PHPLIB-418',
+    ];
+
+    /**
+     * Assert that the expected and actual command documents match.
+     *
+     * Note: this method may modify the $expected object.
+     *
+     * @param stdClass $expected Expected command document
+     * @param stdClass $actual   Actual command document
+     */
+    public static function assertCommandMatches(stdClass $expected, stdClass $actual)
+    {
+        /* The command may contain a session id, even though we don't start one.
+         * In this case, ignore any session identifier by removing it. */
+        if (isset($actual->lsid)) {
+            unset($actual->lsid);
+        }
+
+        static::assertDocumentsMatch($expected, $actual);
+    }
+
+    /**
+     * Execute an individual test case from the specification.
+     *
+     * @dataProvider provideTests
+     * @param string    $name           Test name
+     * @param stdClass  $test           Individual "tests[]" document
+     * @param array     $runOn          Top-level "runOn" array with server requirements
+     * @param array     $data           Top-level "data" array to initialize collection
+     * @param string    $databaseName   Name of database under test
+     * @param string    $collectionName Name of collection under test
```

Comment:
I realize this was copy-pasted from another file, but you can remove a space between the type and variable names. Looks like a missed that in the other files, but we can address it here.

Reply:
Will do. I'll let phpcs clean that up in other files once we add it to the build pipeline.